include/oneapi/dpl/pstl/memory_ranges_impl.h

```diff
@@ -0,0 +1,199 @@
+std::ranges::uninitialized_move_result<std::ranges::borrowed_iterator_t<_InRange>,
+                                       std::ranges::borrowed_iterator_t<_OutRange>>
+__pattern_uninitialized_move(_Tag __tag, _ExecutionPolicy&& __exec, _InRange&& __in_r, _OutRange&& __out_r)
+{
+    assert(std::ranges::size(__in_r) == std::ranges::size(__out_r));
+
+    const auto __first1 = std::ranges::begin(__in_r);
+    const auto __first2 = std::ranges::begin(__out_r);
+    const auto __size = std::ranges::size(__in_r);
+
+    const auto __last1 = __first1 + __size;
+    const auto __last2 = __first2 + __size;
+
+    oneapi::dpl::uninitialized_move(std::forward<_ExecutionPolicy>(__exec), __first1, __last1, __first2);
+
+    return {__last1, __last2};
+}
+
+template<typename _ExecutionPolicy, typename _InRange, typename _OutRange>
+std::ranges::uninitialized_move_result<std::ranges::borrowed_iterator_t<_InRange>,
+                                       std::ranges::borrowed_iterator_t<_OutRange>>
+__pattern_uninitialized_move(__serial_tag</*IsVector*/std::false_type>, _ExecutionPolicy&&, _InRange&& __in_r, _OutRange&& __out_r)
+{
+    return std::ranges::uninitialized_move(std::forward<_InRange>(__in_r), std::forward<_InRange>(__out_r));
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+// __pattern_uninitialized_fill
+//---------------------------------------------------------------------------------------------------------------------
+
+template <typename _Tag, typename _ExecutionPolicy, typename _R, typename _T>
+std::ranges::borrowed_iterator_t<_R>
+__pattern_uninitialized_fill(_Tag __tag, _ExecutionPolicy&& __exec, _R&& __r, const _T& __value)
+{
+    const auto __first = std::ranges::begin(__r);
+    const auto __last = __first + std::ranges::size(__r);
+
+    oneapi::dpl::uninitialized_fill(std::forward<_ExecutionPolicy>(__exec), __first, __last, __value);
+
+    return {__last};
+}
+
+template <typename _ExecutionPolicy, typename _R, typename _T>
+std::ranges::borrowed_iterator_t<_R>
+__pattern_uninitialized_fill(__serial_tag</*IsVector*/ std::false_type>, _ExecutionPolicy&&, _R&& __r, const _T& __value)
+{
+    return std::ranges::uninitialized_fill(std::forward<_R>(__r), __value);
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+// pattern_destroy
+//---------------------------------------------------------------------------------------------------------------------
+
+template <typename _Tag, typename _ExecutionPolicy, typename _R>
+std::ranges::borrowed_iterator_t<_R>
+__pattern_destroy(_Tag __tag, _ExecutionPolicy&& __exec, _R&& __r)
+{
+    const auto __first = std::ranges::begin(__r);
+    const auto __last = __first + std::ranges::size(__r);
+
+    oneapi::dpl::destroy(std::forward<_ExecutionPolicy>(__exec), __first, __last);
+
+    return {__last};
```

Comment:
```suggestion
    return __last;
```

Reply:
Closing it as discussed in https://github.com/uxlfoundation/oneDPL/pull/2292#discussion_r2140170434.